Polygon_repair_2/include/CGAL/Polygon_repair_2/Polygon_repair_2.h

```diff
@@ -0,0 +1,346 @@
+    }
+  }
+
+  // Label triangles in triangulation
+  void label_triangulation() {
+    for (auto const face: t.all_face_handles()) {
+      face->label() = 0;
+      face->processed() = false;
+    }
+
+    // Label exterior with label -1, marking it as processed and
+    // putting interior triangles adjacent to it in to_check
+    std::list<typename Triangulation::Face_handle> to_check;
+    std::list<int> to_check_added_by;
+    label_region(t.infinite_face(), -1, to_check, to_check_added_by);
+
+    // Label region of front element to_check list
+    while (!to_check.empty()) {
+
+      if (to_check.front()->label() == 0) { // label = 0 means not labelled yet
+        if (to_check_added_by.front() < 0) {
+          label_region(to_check.front(), number_of_polygons+1, to_check, to_check_added_by);
+          ++number_of_polygons;
+        } else {
+          label_region(to_check.front(), -(number_of_holes+2), to_check, to_check_added_by);
+          ++number_of_holes;
+        }
+      } to_check.pop_front();
+      to_check_added_by.pop_front();
+
+    } // std::cout << number_of_polygons << " polygons with " << number_of_holes << " holes in triangulation" << std::endl;
+  }
+
+  // Reconstruct ring boundary starting from an edge (face + opposite vertex) that is part of it
+  void reconstruct_ring(std::list<typename Kernel::Point_2>& ring,
+                        typename Triangulation::Face_handle face_adjacent_to_boundary,
+                        int opposite_vertex) {
+
+    // Create ring
+    typename Triangulation::Face_handle current_face = face_adjacent_to_boundary;
+    int current_opposite_vertex = opposite_vertex;
+    do {
+      typename Triangulation::Vertex_handle pivot_vertex = current_face->vertex(current_face->cw(current_opposite_vertex));
+      // std::cout << "Adding point " << pivot_vertex->point() << std::endl;
+      ring.push_back(pivot_vertex->point());
+      typename Triangulation::Face_circulator fc = t.incident_faces(pivot_vertex, current_face);
+      do {
+        ++fc;
+      } while (fc->label() != current_face->label());
+      current_face = fc;
+      current_opposite_vertex = fc->cw(fc->index(pivot_vertex));
+    } while (current_face != face_adjacent_to_boundary ||
+             current_opposite_vertex != opposite_vertex);
+
+    // Start at lexicographically smallest vertex
+    typename std::list<typename Kernel::Point_2>::iterator smallest_vertex = ring.begin();
+    for (typename std::list<typename Kernel::Point_2>::iterator current_vertex = ring.begin();
+         current_vertex != ring.end(); ++current_vertex) {
+      if (*current_vertex < *smallest_vertex) smallest_vertex = current_vertex;
+    } if (ring.front() != *smallest_vertex) {
+      ring.splice(ring.begin(), ring, smallest_vertex, ring.end());
+    }
+  }
+
+  void compute_hole_nesting() {
+    for (int i = 0; i < number_of_holes; ++i) {
+      nesting.emplace_back();
+    } for (auto const &face: t.finite_face_handles()) {
```

Comment:
Do not start with the `for` after the closing `}` but put a newline for readability.  It is ok for a `do { ..} while(..);`